examples/src/tools/hedera.ts

```diff
@@ -0,0 +1,82 @@
+import { 
+  HederaAccountBalance,
+  HederaTransfer,
+  HederaCreateAccount,
+  HederaDeleteAccount,
+} from "@langchain/community/tools/hedera";
+import { PrivateKey } from "@hashgraph/sdk"
+import { StructuredTool } from "@langchain/core/tools";
+import { ChatOpenAI } from "@langchain/openai";
+import type { ChatPromptTemplate } from "@langchain/core/prompts";
+import { createOpenAIFunctionsAgent, AgentExecutor } from "langchain/agents";
+import { pull } from "langchain/hub";
+
+export async function run() {
+    const llm = new ChatOpenAI({
+        model: "gpt-3.5-turbo",
+        temperature: 0,
+      });
+    
+    // Initalize paramters for hedera tools to use
+    const hederaParams = {
+        credentials: {
+          accountId: process.env.HEDERA_ACCOUNT_ID,
+          privateKey: process.env.HEDERA_PRIVATE_KEY,
+        },
+        network: "testnet",
+        maxTransactionFee: 100,
+        maxQueryPayment: 50,
+    };
+
+    // Provide the hedera tools to be used
+    const tools: StructuredTool[] = [
+      new HederaAccountBalance(hederaParams),
+      new HederaTransfer(hederaParams),
+      new HederaCreateAccount(hederaParams),
+      new HederaDeleteAccount(hederaParams),
+    ];
+
+    // Setup the agent to use the hedera tool
+    const prompt = await pull<ChatPromptTemplate>(
+        "hwchase17/openai-functions-agent"
+      );
+      
+      const agent = await createOpenAIFunctionsAgent({
```

Comment:
Let's use `createReactAgent` from `@langchain/langgraph` here instead

Reply:
@jacoblee93 would it be possible to use ```createStructuredChatAgent``` instead?

Reply:
No, let's show the latest recommended practices